lua/outline/highlight.lua

```diff
@@ -9,13 +9,17 @@ local M = {
 ---Clear all highlights in buffer
 ---@param bufnr integer
 function M.clear_all_ns(bufnr)
-  vim.api.nvim_buf_clear_namespace(bufnr, -1, 0, -1)
+  if vim.api.nvim_buf_is_valid(bufnr) then
+    pcall(function() vim.api.nvim_buf_clear_namespace(bufnr, -1, 0, -1) end)
```

Comment:
Is this `pcall` still needed after we've already checked the `bufnr` is valid? You haven't added the pcall anywhere else. Do you still encounter this issue if we unwrap the `clear_namespace` call from pcall here?

Reply:
Hi,
  Yes, as you correctly assumed it is due to race conditions. `pcall` here is needed. I also found another issue with the `Sidebar:update_cursor_pos`. Apparently when we quickly close the outline window right after pressing enter on the selected symbol, the `Sidebar:destroy` is called which sets self.view to nil and then in the `Sidebar:update_cursor_pos` we get an error due to this. 

I think to tackle all these issue we should implement some sort of state machine which only destroys the objects when we are done with all the callbacks. What are your thoughts on this ?

Reply:
FYI, to reproduce the issue: 
When the PC is under some load, 
   - Quickly open the symbol sidebar select a symbol (which is known to be bit lower in the file) 
   - Press enter on the selected symbol and quickly close the outline sidebar.

I see a lot of errors popping up when I do this. When this happens, outline auto-commands are left in some inconsistent state. Therefore on every cursor move similar errors keeps popping up. I then have to re-open the neovim to fix it.